### ISSUE TITLE
Error - The specified CGI application encountered an error and the server terminated the process.

### ISSUE THREAD

**Issue by ankurjasoria** [aspnet/AspNetCore#5911]
Hi All I am facing issue "The specified CGI application encountered an error and the server terminated the process." after hosting it on Azure Web Sites. On Local it works fine. I have developed application on asp.net core 1.0 and deployed on Azure.

Is there any timeout issue or any thing missing when deployed code on azure ?
Please help me out regarding this issue.

**Comment by muratg** [comment]
No more action is planned on this issue so I'm closing this. Folks can check the troubleshooting guide linked above for similar issues.

**Comment by ps2goat** [comment]
FYI, we ran into this issue today with the CGI application error.

## TL;DR;
Azure dev ops or `dotnet publish` seems to be doing XML transforms for us on our web.config files, and we ended up having two entries in the web.config for `ASPNETCORE_ENVIRONMENT`.  

The ultimate source for discovering this was [here](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2#setting-environment-variables)

> Warning
> 
> Environment variables set in this section conflict with system environment variables set with the same name. If an environment variable is set in both the web.config file and at the system level in Windows, the value from the web.config file becomes appended to the system environment variable value (for example, ASPNETCORE_ENVIRONMENT: Development;Development), which prevents the app from starting.

## Details
We had the following in `web.Release.config`, and it was normally transformed by Octopus Deploy when deploying.

```
<?xml version="1.0" encoding="utf-8"?>
<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
  <system.webServer>
    <aspNetCore>
      <environmentVariables xdt:Transform="Insert">
        <environmentVariable name="ASPNETCORE_ENVIRONMENT" value="#{HostingEnvironment}" />
      </environmentVariables>
    </aspNetCore>
  </system.webServer>
</configuration>
```

Somewhere in our Azure Dev Ops build, the transform was already taking place, but, the transform file was still in the artifacts list which was then placed into our Nuget package.  That Nuget package was then shipped to Octopus Deploy, which did another transform.  Because we used `Insert` for the transform command, a second copy of the environment variable was inserted into the `web.config` file.

This is a recent change.  This project is still on dotnet core 1.1, I believe, and we just had a successful build and deploy on 1/30/2019.  A rebuild of the exact same code base also caused this problem, and it doesn't appear to be an Octopus change.

**Comment by arunpp1987** [comment]
Mates, I was able to solve this issue by removing forwardWindowsAuthToken from the web.config file under wwwroot. Please do the following steps.

Step 1)Navigate to src/ProjectName/wwwroot
Step 2)Open the web.config
Step 3)In the httpPlatformremove the forwardWindowsAuthToken="true/false" property
Step 4)Redeploy and mine worked fine.
